xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecText.cpp

```diff
@@ -9,31 +9,47 @@
 #include "DVDOverlayCodecText.h"
 
 #include "DVDCodecs/DVDCodecs.h"
-#include "DVDOverlayText.h"
+#include "DVDOverlayASS.h"
 #include "DVDStreamInfo.h"
+#include "Util.h"
 #include "cores/VideoPlayer/DVDSubtitles/DVDSubtitleTagSami.h"
 #include "cores/VideoPlayer/Interface/DemuxPacket.h"
+#include "cores/VideoPlayer/Interface/TimingConstants.h"
+#include "utils/StringUtils.h"
 #include "utils/log.h"
 
+#include <memory>
+
 #include "system.h"
 
-CDVDOverlayCodecText::CDVDOverlayCodecText() : CDVDOverlayCodec("Text Subtitle Decoder")
+CDVDOverlayCodecText::CDVDOverlayCodecText()
+  : CDVDOverlayCodec("Text Subtitle Decoder"), m_adapter(std::make_shared<CSubtitlesAdapter>())
 {
   m_pOverlay = NULL;
+  m_output = false;
+  m_bIsSSA = false;
 }
 
 CDVDOverlayCodecText::~CDVDOverlayCodecText()
 {
-  if(m_pOverlay)
-    SAFE_RELEASE(m_pOverlay);
+  Dispose();
 }
 
 bool CDVDOverlayCodecText::Open(CDVDStreamInfo &hints, CDVDCodecOptions &options)
 {
-  if (hints.codec == AV_CODEC_ID_TEXT || hints.codec == AV_CODEC_ID_SUBRIP)
-    return true;
+  bool supportedCodec = false;
+  if (hints.codec == AV_CODEC_ID_TEXT || hints.codec == AV_CODEC_ID_SSA)
+    supportedCodec = true;
+  if (hints.codec == AV_CODEC_ID_SUBRIP)
+    supportedCodec = true;
+  if (!supportedCodec)
+    return false;
 
-  return false;
+  m_bIsSSA = (hints.codec == AV_CODEC_ID_SSA);
```

Comment:
Please check if you really need to store the state. You can simply do a comparison against the AV_CODEC_ID_SSA where needed, which to me looks like a single place

Reply:
Ok, then I think that storing the codec and doing a further comparison is better than having a member Boolean completely tied to a single codec type

Reply:
done